unified-codes/apps/data-service/src/app/v1/types/DgraphDataSource.ts

```diff
@@ -333,34 +339,77 @@ export class DgraphDataSource extends RESTDataSource {
               route.children.forEach(doseForm => {
                 if (doseForm.children) {
                   doseForm.children.forEach(doseQualifierOrDoseStrength => {
-                      if (doseQualifierOrDoseStrength.children) {
-                        doseQualifierOrDoseStrength.children.forEach(doseStrength => {
+                      if (doseQualifierOrDoseStrength.type === EEntityTypeV2.DoseStrength) {
```

Comment:
This works well so I think all good for now! May be nice later to banish the mSupply specific stuff to its own file 🧙